### ISSUE TITLE
Auto detection of language when writing posts/comments.

### ISSUE THREAD

**Issue by lullis** [LemmyNet/lemmy-ui#1231]
### Is your proposal related to a problem?

Having to select which language to use on every post is an unnecessary pain. ~The list is also unnecessary long~ (Ignore, if you select the languages the list is filtered down)

### Describe the solution you'd like.

~First, I'd like to be able to indicate which languages I can actually use. Maybe this requires some work on the backend to save the settings, but it could be done client-side by reading the values from the browser settings.~ 

I'd also like to have a sensilble *default* language: If I'm responding to a post in Portuguese, it's quite likely that my answer will be in Portuguese as well.

### Describe alternatives you've considered.

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/i18n/detectLanguage

**Comment by pproe** [comment]
Would be useful to include a "Preferred Languages" list as part of user settings. If only one language is included, simply apply that for each post. Else, offer a concise list of all preferred languages for the user to select.

**Comment by BCNelson** [comment]
this would probably be useful https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/i18n/detectLanguage

**Comment by BCNelson** [comment]
This could be used https://github.com/kwonoj/cld3-asm in the browser

**Comment by loid345** [comment]
You can write in any language if there is a built-in translator like in Facebook linkedin or soapbox

![image](https://github.com/LemmyNet/lemmy-ui/assets/20200970/3def1468-006f-4767-be37-de81edc9c831)

**Comment by Zetaphor** [comment]
In the very least we should be selecting a default when only one option is available.
I'm attempting to make a post on lemmy.world in a thread for which English is the only available option, but if I try and submit without first selecting a language my submission fails.

I've also seen this issue occurring in Jerboa

**Comment by KaKi87** [comment]
Hello,

I would like to suggest using by default :
- when creating a post, the last created post's language ;
- when commenting a post, the post's language ;
- when replying to a comment, the comment's language.

Thanks

**Comment by tripathiakshit** [comment]
A simpler implementation could just be to have an option for default comment language in the user's profile settings, and applying that instead of undetermined.

**Comment by KaKi87** [comment]
Not really, because many of us write in multiple languages, not determined by our nationality, but by the community we're posting in, the author of the post we're replying to, etc.

Even if there's a language one writes in more than another, they can often switch from one to the other in the same minute, just because they're interacting in different communities, and that's when the user would forget to change the selection, which regularly happens on Mastodon already.

**Comment by dessalines** [comment]
Out of scope for us. If someone wants to add this, I'll re-open and assign it to you.
